### PR TITLE
Update to Laravel 5.8, new PDF generator, Fix Excel export (WIP)

### PR DESCRIPTION
**Pull request by CristianDeluxe**
I created a new Laravel project and copied file by file from Attendize, integrating, updating and merging.

It's a big change (a lot of files) but I think it was necessary to start as clean as possible.

You can follow the changes more simply by looking at my commits separately.

- Updated all to 5.7 and tested
- Deleted unmaintained package from 2017 nitmedia/wkhtml2pdf
- The pdfs for the tickets are now generated by the brand new barryvdh/laravel-dompdf, no more permission errors or binaries needed.
- Updated maatwebsite/excel to v3 and fixed Excel export not working **(still in process)**
- Updated laracasts/utilities to v3
- Now all routes use the new Laravel 5.7 syntax, which has allowed to discover an error in the namespace of API routes.
- The helpers have been rewritten and now use ServiceProviders
- Deleted some Service Providers and Aliases from config/app.php file because from laravel 5.5 some have activated the auto discovery function.
- Fixed some tests, although some functions are not available, I think it is not worth spending more time as they need to be rewritten from scratch.

### Breaking Changes
- Now by default is used utf8mb4 encoding that supports emojis and comes with Laravel 5.7
- The images in the events table should be renamed from assets/images/.. to images/...
- Laravel by default tries to write in the database when the field is empty, this is causing errors in the validation, because, for example, when editing an organizer, if we do not fill in the Facebook field gives an error, we must solve it in the code or marking these fields in the database as nullables.

I suggest to create a new branch for these changes. Perhaps 2.0? since it will be necessary to perform migration codes for the old installations.

### CODE REVIEW

**Comment by jeremyquinton** [comment]
@CristianDeluxe just some feedback. The PR has many great things however I can't merge it as its huge. 

Are you able to break it down into smaller PRS. For example the PDF stuff is really useful and that would be merge instantly. 

We need to be incremental with small changes than a big bang approach.

**Comment by CristianDeluxe** [comment]
> @CristianDeluxe just some feedback. The PR has many great things however I can't merge it as its huge.
> 
> Are you able to break it down into smaller PRS. For example the PDF stuff is really useful and that would be merge instantly.
> 
> We need to be incremental with small changes than a big bang approach.

Hi, sorry I've been so busy lately.

I really don't know how I can do now to "chop" the changes, I know that there are enough changes, so I proposed to start a branch v2.0, so we could test everything quietly, create new tests and forget version 1 in a few months.

**Comment by publicarray** [comment]
> I really don't know how I can do now to "chop" the change

If you are on a Mac I would use gitup https://gitup.co/ as a GUI to help me split the commits into seperate branches (one for each feature) i.e create a new branch from upstream (this repos) master and “cherry pick” the commits for a particular feature into the new branch. If there is a commit where you only want part of the changes. I would still “cherry pick” it and then “edit” (“edit” in GitUp/“rebase” in git) the commit

Hope this helps

**Comment by mrlexley** [comment]
I think there are great changes in it, I hope you can take the time to make them compatible with the current master :-) specialy the PDF generator

**Comment by justynpride** [comment]
Any update on being able to split and merge these in?

File: package.json [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+        "jquery": "^3.2",
+        "laravel-elixir": "*",
+        "laravel-mix": "^2.0",
+        "lodash": "^4.17.5",
+        "popper.js": "^1.12",
+        "vue": "^2.5.17"
+    },
+    "scripts": {
+        "dev": "npm run development",
+        "development": "cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js",
```
[the remark this answers is not in the document]
Comment (CristianDeluxe) [reply package.json]: Hello, this command is "copied" from the original Laravel file.

**Comment by CristianDeluxe** [comment]
Hi guys, sorry, for work and personal reasons until now I have not been able to work on this.

I have updated the repository to synchronize it with the current changes and I have taken the opportunity to update it to version 5.8.

I understand that this PR cannot be merged as it is, but: How can we include the changes?

As I suggested previously, it would be perfect to start an Attendize 2.0 branch to include all these changes and be able to start with all the tests.

**Comment by zenny** [comment]
@CristianDeluxe This looks very promising! Great job!

**Comment by CristianDeluxe** [comment]
I close this PR so as not to cause confusion, since most of the changes have been implemented in two other PRs.

**Pull request closed without merging**